src/shared/settingsConfiguration.ts

```diff
@@ -5,231 +5,590 @@
 
 import * as vscode from 'vscode'
 import * as packageJson from '../../package.json'
-import { ClassToInterfaceType } from './utilities/tsUtils'
-import { isAutomation, isReleaseVersion } from './vscode/env'
-import * as logger from './logger'
-import globals from './extensionGlobals'
+import { getLogger } from './logger'
+import { cast, FromDescriptor, TypeConstructor, TypeDescriptor } from './utilities/typeConstructors'
+import { ClassToInterfaceType, keys } from './utilities/tsUtils'
+import { toRecord } from './utilities/collectionUtils'
+import { isAutomation } from './vscode/env'
+
+type Workspace = Pick<typeof vscode.workspace, 'getConfiguration' | 'onDidChangeConfiguration'>
+type ScopedEmitter = vscode.EventEmitter<vscode.ConfigurationChangeEvent>
 
 /**
- * Wraps the VSCode configuration API and provides Toolkit-related
- * configuration functions.
+ * A class for manpilulating settings shared by all extensions.
```

Comment:
The reason for mentioning vscode in some of the docstrings, is to make it clear that this isn't about AWS settings, service settings, etc. This helps new contributors.

```suggestion
 * A class for manipulating VSCode user settings (from all extensions).
```